doc/hcr.rst

```diff
@@ -98,34 +99,40 @@ To install SDL2 you can use `nimble install sdl2`.
   main()
 
 
-Compile this example via::
+Compile this example via:
 
+```cmd
   nim c --hotcodereloading:on mymain.nim
+```
 
 Now start the program and KEEP it running!
 
-::
-
+.. code:: cmd
   # Unix:
   mymain &
   # or Windows (click on the .exe)
   mymain.exe
   # edit
 
-For example, change the line::
+For example, change the line:
 
+```nim
```

Comment:
so when can i use 

````
```nim
...
```
````

(which brings us closer to markdown, which is good) ? can any instead of
```
.. code-block:: cmd
  some code...
```
be replaced or are there conditions?

Reply:
always, they are equivalent. You can write \`\`\`cmd and \`\`\`nim and any other language.

However note bug 22 from #17340. When it's resolved we can convert to using this Markdown style as a unified style in this repo just as you wanted.

There is 1 thing that I don't like about Markdown style: addition of those spaces at the beginning of lines. It would be good to visually distinguish code lines in documentation by additional indentation without adding them into final document.

I'd prefer to **pass the code block between \`\`\` through `dedent`**. WDYT?

Reply:
Actually I found in the specification

https://spec.commonmark.org/0.29/#fenced-code-block

>  If the leading code fence is indented N spaces, then up to N spaces of indentation are removed from each line of the content (if present).

And this does actually work:

````
  ```
  some code
  ```
````

  ```
  some code
  ```

Reply:
cool;

can you explain 22?
this seems to render fine with nim doc, doesn't it?
````nim
proc main*()=
  ##[
  foo1:
  ```nim
  echo 1
  ```
  * foo2:
    ```nim
    echo 2
    ```
  ]##

main()
````

Reply:
add two additional spaces as indentation before `echo 1` or `echo 2`.

Reply:
filed bug 23 in the same place :-)